Source/Core/Core/IOS/Network/Socket.h

```diff
@@ -197,14 +199,21 @@ class WiiSocket
   s32 CloseFd();
   s32 FCntl(u32 cmd, u32 arg);
 
+  bool HasTimeout();
+  void ResetTimeout();
+
   void DoSock(Request request, NET_IOCTL type);
   void DoSock(Request request, SSL_IOCTL type);
   void Update(bool read, bool write, bool except);
   bool IsValid() const { return fd >= 0; }
+
   s32 fd = -1;
   s32 wii_fd = -1;
   bool nonBlock = false;
   std::list<sockop> pending_sockops;
+
+  using timeout_t = std::chrono::time_point<std::chrono::system_clock>;
```

Comment:
I don't think introducing this type alias is necessary as it doesn't seem to be used. Also the _t suffix is reserved for POSIX stuff so I would avoid using it.